### DISCUSSION TITLE
a question in 3D navier stokes

### DISCUSSION THREAD

**Discussion by Oops-Qiao** [idaholab/moose#22650] [Q&A General] [answered]
Hello,

I have a simple question when I look at your moose code regarding the velocity component.
https://github.com/idaholab/moose/blob/next/modules/navier_stokes/src/kernels/NSMomentumInviscidFlux.C
In line 42 we have
  // For _component = k,
  // (rho*u) * u_k = (rho*u_k) * u <- we write it this way
  RealVectorValue vec(_u[_qp] * _u_vel[_qp],  // (U_k) * u_1
                      _u[_qp] * _v_vel[_qp],  // (U_k) * u_2
                      _u[_qp] * _w_vel[_qp]); // (U_k) * u_3

and 
https://github.com/idaholab/moose/blob/next/modules/navier_stokes/src/kernels/NSEnergyInviscidFlux.C
in line 40
  // velocity vector
  RealVectorValue vel(_u_vel[_qp], _v_vel[_qp], _w_vel[_qp]);

I just wonder that in 3D geometry, should each component of velocity be defined as a vector? It looks to me that in the upper two examples in your code, the velocity component is just a real such that the vel(_u_vel[_qp], _v_vel[_qp], _w_vel[_qp]) is a vector.

It is a little bit confusing. Thank you for clarification.

Best regards
Qia.

**Comment by Oops-Qiao** [comment, marked as answer]
@GiudGiud
Hello, :) There seems to be an interesting question.

**Reply by GiudGiud** [reply]
no each component is not a vector. 
What you are seeing is a momentum flux so you have u * u terms

**Reply by Oops-Qiao** [reply]
Thank you so much.
